x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/pivot/Pivot.java

```diff
@@ -142,17 +136,17 @@ public void preview(
                     final Aggregations aggregations = r.getAggregations();
                     if (aggregations == null) {
                         listener.onFailure(
-                            new ElasticsearchStatusException("Source indices have been deleted or closed.", RestStatus.BAD_REQUEST)
-                        );
+                            new ElasticsearchStatusException("Source indices have been deleted or closed.", RestStatus.BAD_REQUEST));
                         return;
                     }
                     final CompositeAggregation agg = aggregations.get(COMPOSITE_AGGREGATION_NAME);
                     TransformIndexerStats stats = new TransformIndexerStats();
-                    // remove all internal fields
 
-                    List<Map<String, Object>> docs = extractResults(agg, fieldTypeMap, stats).peek(
-                        doc -> doc.keySet().removeIf(k -> k.startsWith("_"))
-                    ).collect(Collectors.toList());
```

Comment:
Ah, I see the pivot was already doing this. But it is still a concern. Especially since now as we are copying the doc directly.

Reply:
Needs investigation, added TODO in the `DocumentConversionUtils.removeInternalFields` method comment.